projects/igniteui-angular/src/lib/navigation-drawer/themes/_base.scss

```diff
@@ -0,0 +1,265 @@
+
+            igx-icon {
+                width: var(--igx-icon-size, rem(24px));
+                height: var(--igx-icon-size, rem(24px));
+                font-size: var(--igx-icon-size, rem(24px));
+                margin: 0;
+            }
+
+            &:hover,
+            &:focus {
+                box-shadow: none;
+                border: none;
+            }
+        }
+
+        &%igx-button--fab {
+            min-height: auto;
+        }
+
+        &%igx-icon-button-display {
+            height: auto;
+            transition: none;
+        }
+
+        &:hover,
+        &:focus {
+            background: var-get($_theme, 'item-hover-background');
+            color: var-get($_theme, 'item-hover-text-color');
+            box-shadow: none;
+
+            igx-icon {
+                color: var-get($_theme, 'item-hover-icon-color');
+            }
+        }
+    }
+
+    %active {
+        color: var-get($_theme, 'item-active-text-color');
+        background: var-get($_theme, 'item-active-background');
+
+        igx-icon {
+            color: var-get($_theme, 'item-active-icon-color');
+        }
+    }
+
+    @include b(igx-nav-drawer) {
+        --igx-nav-drawer-size: #{rem(240px)};
+
+        flex-basis: 0;
+        transition: flex-basis;
+        transition-duration: .3s;
+        transition-timing-function: $ease-out-quad;
+        flex-shrink: 0;
+
+        @include m(pinned) {
+            flex-basis: var(--igx-nav-drawer-size);
+        }
+
+        @include mx(mini, pinned) {
+            flex-basis: calc(var(--igx-mini-nav-drawer-size, #{var(--igx-nav-drawer-size--mini)}) + rem(1px));
+        }
+
+        @include e(aside) {
+            position: fixed;
+            height: 100%;
+            min-height: 100%;
+            overflow-x: hidden;
+            background: var-get($_theme, 'background');
+            top: 0;
+            bottom: 0;
+            width: var(--igx-nav-drawer-size);
+            inset-inline-start: 0;
+            z-index: 999;
+            transition: width, padding, transform;
+            transition-timing-function: $ease-in-out-quad;
+            box-shadow: var-get($_theme, 'elevation');
+            padding: var(--aside-padding);
+            border-inline-end: rem(1px) solid var-get($_theme, 'border-color');
+            border-radius: var-get($_theme, 'border-radius');
+
+            &.panning {
+                overflow-x: hidden;
+                transition: none;
+            }
+        }
+
+        @include e(aside, collapsed) {
+            transform: translate3d(-300px, 0, 0);
+
+            [dir='rtl'] & {
+                transform: translate3d(300px, 0, 0);
+            }
+
+            box-shadow: none;
+
+            &.igx-nav-drawer__aside--right {
+                transform: translate3d(300px, 0, 0);
+                box-shadow: none;
+
+                [dir='rtl'] & {
+                    transform: translate3d(-300px, 0, 0);
+                }
+            }
```

Comment:
If you want to apply some style rules when the aside is both collapsed and right aligned, you should use the $mods syntax:

```scss
@include e(aside, $mods: (collapsed, right) {
/* ... style rules */
}
```